File: Java/프로그래머스/3/161988. 연속 펄스 부분 수열의 합/연속 펄스 부분 수열의 합.java
class Solution {
    public long solution(int[] sequence) {
        long answer = 0;
        boolean isPositive = true;
        long sum1 = 0;
        long sum2 = 0;

        for (int i : sequence) {
            sum1 += isPositive ? i : -i;
            sum2 += isPositive ? -i : i;

            sum1 = Math.max(sum1, 0);
            sum2 = Math.max(sum2, 0);

            answer = Math.max(answer, Math.max(sum1, sum2));
            isPositive = !isPositive;
        }
        return answer;
    }
}

File: Python/프로그래머스/3/161988. 연속 펄스 부분 수열의 합/연속 펄스 부분 수열의 합.py
def solution(sequence):
    answer = 0
    is_positive = True
    sum1, sum2 = 0, 0

    for i in sequence:
        sum1 += i if is_positive else -i
        sum2 += -i if is_positive else i

        sum1 = max(sum1, 0)
        sum2 = max(sum2, 0)

        answer = max(answer, max(sum1, sum2))
        is_positive = not is_positive

    return answer